lib/pacer/pacer.go

```diff
@@ -309,6 +317,42 @@ func (p *Pacer) drivePacer(retry bool) {
 	}
 }
 
+// s3Pacer implements a pacer compatible with our expectations of S3, where it tries to not
+// delay at all between successful calls, but backs off in the default fashion in response
+// to any errors.
+// The assumption is that errors should be exceedingly rare (S3 seems to have largely solved
+//  the sort of scability questions rclone is likely to run into), and in the happy case
+//  it can handle calls with no delays between them.
+//
+// Basically defaultPacer, but with some handling of sleepTime going to/from 0ms
+// Ignores minSleep entirely
+//
+// Call with p.mu held
+func (p *Pacer) s3Pacer(retry bool) {
+	oldSleepTime := p.sleepTime
+	if retry {
+		if p.attackConstant == 0 {
+			p.sleepTime = p.maxSleep
+		} else {
+			if p.sleepTime == 0 {
+				p.sleepTime = 1
```

Comment:
I think this should be p.sleepTime = p.minSleep

I think minSleep should be something like `10ms` for the amazon pacer so we sleep for 10ms minimum on an error, and that is increased exponentially.

The constant `1` is 1 nS so not a lot of sleeping, and still only 1 second after 30 retries!